docs/modules/ROOT/pages/explanations/decisions/crossplane-impl.adoc

```diff
@@ -0,0 +1,185 @@
+This includes cloud instances provisioned by Crossplane itself.
+
+Additional logic is implemented through custom webhooks and controllers.
+
+.Service provisioning with an umbrella Helm chart
+image::appcat_proposal_1.drawio.svg[]
+
+The "umbrella" chart contain all the feature gates and dependent charts required to provision services on the supported cloud infrastructure.
+The chart takes input parameters such as
+
+- cloud provider
+- major version
+- backup enabled
+- SLA (alerting enabled)
+
+and then deploys the resources as required.
+
+The chart is also used when cloud instances are provisioned.
+For example, there may be additional Prometheus exporters needed in case the cloud instance doesn't provide Prometheus metrics endpoints.
+
+In addition to the composition, there are webhooks to perform validations and assignment to customer.
+Also, a dedicated version update controller delays changing of `CompositionRevisions` to the newest version until a specific time is met.
+
+Advantages::
+- Umbrella chart is able to deploy arbitrary resources
+- Umbrella chart is integration testable
+
+Disadvantages::
+- Separate version controller required for automated updates.
+- Webhook controller required (possibly per service).
+- Logic and feature gates in umbrella chart is limited by Helm's templating engine.
+
+=== Proposal 2: Side Charts
+
+The main difference to proposal 1 is that the umbrella chart is split up into parts where just the service is deployed and another where the additional resources are deployed.
+
+It is assumed that the additional resources are roughly the same for all services, thus they are packaged into one side chart.
+
+.Service provisioning with a common "side" Helm chart
+image::appcat_proposal_2.drawio.svg[]
+
+If the service is a cloud instance, instead of deploying the service chart, the Crossplane resource is used.
+
+In addition to the composition, there are webhooks to perform validations and assignment to customer.
+Also, a dedicated version update controller delays changing of `CompositionRevisions` to the newest version until a specific time is met.
+
+Advantages::
+- Easier onboarding of new services.
+- Side chart is able to deploy arbitrary resources.
+
+Disadvantages::
+- It assumes that every service requires roughly the same common resources that can be packed into one chart.
+- Webhook controller required (possibly per service).
+- Logic and feature gates in side chart is limited by Helm's templating engine.
+- Side chart and service definition are loosely coupled.
+
+=== Proposal 3: Dedicated Provider
+
+This proposal uses a dedicated Crossplane provider to deploy and configure all the resources that the service needs.
+A Crossplane provider uses code thus provides the most flexibility.
+
+.Service provisioning with a dedicated Crossplane provider
+image::appcat_proposal_3.drawio.svg[]
+
+Advantages::
+- Flexible deployment using code and Kubernetes API.
+- Webhook controller can be built-in.
```

Comment:
I'm not sure about that one. The providers I've seen so far are only short-lived and are spawned on-demand if changes need to be applied. Maybe it's possible to have always-on providers? If not we'd probably need a seperate controller for still.

Reply:
Webhooks are coming in 1.7 next week: https://github.com/crossplane/crossplane/pull/2919

Reply:
> The providers I've seen so far are only short-lived and are spawned on-demand if changes need to be applied.

What do you mean by this? The Helm provider for example is nothing but an operator that is reconciling Helm releases.

And with "Webhook controller can be built-in." I meant that it's easy to develop/generate a webhook controller that lives in the same code, but maybe runs in a separate pod just like any other Kubernetes Operator using webhooks. Do you feel I should be more specific about this?

Reply:
> Do you feel I should be more specific about this?

Yes please, I think that's the source of my confusion here.